indico/web/forms/fields.py

```diff
@@ -256,10 +256,13 @@ class CategoryField(HiddenField):
 
     def process_formdata(self, valuelist):
         from indico.modules.categories import Category
-        if valuelist:
-            category_id = int(valuelist[0])
+        if valuelist and hasattr(valuelist, 'id'):
```

Comment:
When does `valuelist` have an `id` attribute? This comes from `request.form.getlist('whatever')` so it should always be a plain `list`...